lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
     class DirectObjectReference
       def initialize(object)
         @object = object
       end
 
+      # The objects desription (`#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Not relevant to a direct object reference.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Class
       def target
         @object.class
       end
```

Comment:
Should this be implemented simply as `@object` (no `.class`)? In #832, we discussed how the name `target` was chosen because it returns the target of the reference -- that is, what the reference is a reference to, but in the `DirectObjectReference` case, the reference is to `@object`, not `@object.class`.  OTOH, if the purpose of `target` is to _always_ return a class or module, that's fine, but I think the name needs to reflect that.

In the case of ActiveRecord dynamic attributes (which drove the creation of this, right?) I don't think it matters because object doubles don't have the problem you are solving since `ActiveRecord::Base#respond_to?` always supports dynamic column methods even though `ActiveRecord::Base.method_defined?` doesn't.

Sorry to bring up the naming thing we had settled before but I worry that we'll be annoyed by this inconsistency in the future and it's nice to get it right before 3.2 ships.